src/web-auth/index.js

```diff
@@ -190,31 +188,35 @@ WebAuth.prototype.validateAuthenticationResponse = function(options, parsedHash,
   var transactionNonce = options.nonce || (transaction && transaction.nonce) || null;
 
   var applicationStatus = (transaction && transaction.appStatus) || null;
-  if (parsedHash.id_token && options._idTokenVerification) {
-    return this.validateToken(parsedHash.id_token, transactionNonce, function(
-      validationError,
-      payload
-    ) {
-      if (validationError) {
+
+  if (!parsedHash.id_token) {
+    return cb(null, buildParseHashResponse(parsedHash, applicationStatus, null));
```

Comment:
What is applicationStatus?  And how is it used when there is no id_token?  I can't quite follow this, and it'd be helpful to write some comments that describe what is happening here and why.

Reply:
applicationStatus is something that the user can save in the transaction status. When finding the code to show you, I found an issue talking about a bug related to this: https://github.com/auth0/auth0.js/issues/412
So, we store it as `appState`, but query as `appStatus`. So, effectively, no one is able to use this because it's totally broken. I'll fix in a new commit.

Reply:
fix with a test in https://github.com/auth0/auth0.js/pull/539/commits/75c32a33a2b83299e79419e947ef7f9ae5daa15c